### PR TITLE
Feature/eodhp 994 status tracking

### PR DESCRIPTION
**Pull request by jlangstone-tpzuk**
- Implemented the `consume` cobra CLI application, separate from the main API application.
- Consumes messages from the `workspace-status` topic and updates the database accordingly, i.e. any incoming workspace status changes of relevance to the service are recorded in the database on receipt.
- These changes do not need to have been triggered by a message sent by the workspace service API. I.e. the `workspace-manager` and `workspace-services` are decoupled in that sense.
- A `last_updated` timestamp column is added to the database. If the timestamp of the status is older than the `last_updated` timestamp of the db entry, the message is discard.
- Moved across commmon structs over to `workspace-manager` and importing instead.

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -94,7 +82,8 @@ func (w *WorkspaceDB) InitTables() error {
 				member_group TEXT NOT NULL,
 				role_name TEXT NULL,
 				role_arn TEXT NULL,
-				status TEXT NOT NULL
+				status TEXT NOT NULL,
+				last_updated TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
```
Comment (sgillies-tpzuk) [inline db/db.go:86]: For last_updated, can this cause any race condition errors? E.g. update A (earlier) and update B (later) sent from source, update A gets chosen by a consumer A but takes longer to process, update B is actioned first. Update A would therefore not be processed. Is that definitely the desired behaviour? 

I guess if updates always have the complete state of the workspace then it is fine, just confirming.

**Review by sgillies-tpzuk — COMMENTED** [review]
I think everything looks good, just looking for confirmation surrounding use of last_update time.

File: db/db.go (hunk 1 above)
Comment (jlangstone-tpzuk) [reply db/db.go:86]: That is the desired behavior yes. Whatever is consumed will always be the true reflection of the workspace status at that point in time.

**Review by sgillies-tpzuk — APPROVED** [review]
all good, thanks

**Pull request merged**